src/boot/shim.c

```diff
@@ -90,9 +99,10 @@ EFI_STATUS shim_load_image(
         assert(device_path);
         assert(ret_image);
 
+        // TODO: drop lock protocol and just use plain BS->LoadImage once Shim < 16 is no longer supported
         bool have_shim = shim_loaded();
 
-        if (have_shim)
+        if (have_shim && !shim_loader_available())
```

Comment:
kinda weird you cache_shim_loaded() instead of `(shim_loaded() && !shim_loader_available())`